ospd_openvas/daemon.py

```diff
@@ -1263,9 +1295,19 @@ def exec_scan(self, scan_id: str):
 
         no_id_found = False
         while True:
+            if not self.openvas_process_is_alive(
+                kbdb, ovas_pid, openvas_scan_id
+            ):
+                logger.error(
+                    'Task %s was unexpectedly stopped or killed.', scan_id,
+                )
+                return 1
```

Comment:
Just saw that the `exec_scan(self, scan_id: str):` signature is wrong. It must be `exec_scan(self, scan_id: str) -> int:` and needs a docstring about the meaning of the values but besides that returning an int from a function is not very pythonic. If you need to return some status code it is better to introduce an Enum for this purpose. If values > 0 are errors an exception should be raised instead.

Reply:
I agree, but this is out of the scope of this PR. I will prepare a new PR for this later